src/buyflow/Buyflow.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import AgeStep from './AgeStep'
+import EmailStep from './EmailStep'
+import SummaryStep from './SummaryStep'
+
+interface BuyflowProps {
+  productId: ProductIds
+}
+
+export enum ProductIds {
+  devIns = 'dev_ins',
+}
+
+const PRODUCT_IDS_TO_NAMES = {
+  [ProductIds.devIns]: 'Developer Insurance',
+}
+
+const Buyflow: React.FC<BuyflowProps> = (props) => {
+  const [currentStep, setStep] = useState('email')
+  const [collectedData, updateData] = useState({
+    email: '',
+    age: 0,
+  })
+  const getStepCallback = (nextStep: string) => (field: string, value: any) => {
+    updateData({ ...collectedData, [field]: value })
+    setStep(nextStep)
+  }
+  return (
+    <>
+      <h4>Buying {PRODUCT_IDS_TO_NAMES[props.productId]}</h4>
+      {(currentStep === 'email' && <EmailStep cb={getStepCallback('age')} />) ||
+        (currentStep === 'age' && (
+          <AgeStep cb={getStepCallback('summary')} />
+        )) ||
+        (currentStep === 'summary' && (
+          <SummaryStep collectedData={collectedData} />
+        ))}
+    </>
+  )
+}
+
+export default Buyflow
```

Comment:
looks like refactoring a few main points from this file and keeping structure and step components almost as is, we could achieve most of the tasks:
- [x] we would untangle step components from flow component
- [x] products/flow configuration would become a separate thing that could be easily adapted if needed
- [x] since flow component, step components, and configuration are all separated each one of them would become easier to maintain
- [x] for the same reason new architecture would become easily scalable: we can just increase the number of particular parts and it won't affect everything else

Reply:
P.S.: I probably have missed few things here and there, but my intention was to provide a high-level idea of the future architecture, not the component code as-is. So I would expect a developer to be able to change all the other needed places, cover everything with new tests, and even uncover some new unknown dependency that he would need to discuss with me.

Thanks for your attention, that's everything I have to say about the task at this point.